modules/core/src/smithy4s/Hints.scala

```diff
@@ -97,12 +78,12 @@ object Hints {
       s"Hints(${all.map(_.value).mkString(", ")})"
   }
 
-  case class Binding[A](key: Key[A], value: A) {
-    def tuple: (Key[A], Binding[A]) = key -> this
+  case class Binding[A](key: ShapeTag[A], value: A) {
+    def tuple: (ShapeTag[A], Binding[A]) = key -> this
```

Comment:
if you wanna be fancy... does this work?

```suggestion
    def tuple: (ShapeTag[A], this.type) = key -> this
```

Reply:
608c0c1